index.js

```diff
@@ -0,0 +1,71 @@
+const Docker = require('dockerode');
+
+const wait = (seconds) => new Promise(resolve => setTimeout(resolve, seconds * 1000));
+
+const containers = {};
+
+// log when we go over threshold or when we go back under threshold:
+const logThresholdExceeded = (container, value, options) => {
+  const containerInfo = containers[container.id];
+  if (value < options.cpuThreshold) {
+    if (containerInfo.intervals > options.intervalsAllowed) {
+      console.log(`OKAY: Container ${container.name} CPU is now at ${value}%`);
+    }
+    containerInfo.intervals = 0;
+  } else {
+    containerInfo.intervals ++;
+    if (containerInfo.intervals > options.intervalsAllowed) {
+      console.log(`WARNING: Container ${container.name} has been at ${value}% CPU Usage for ${containerInfo.intervals * options.interval} seconds`);
+    }
+  }
+};
+
+const printStats = (container, stats, options) => {
+  // get cpu usage stats:
+  const cpuStats = stats.cpu_stats;
+  const cpuDelta = cpuStats.cpu_usage.total_usage - containers[container.id].previousCPU;
+  const systemDelta = cpuStats.system_cpu_usage - containers[container.id].previousSystem;
+  const cpuCount = cpuStats.cpu_usage.percpu_usage.length;
+  const cpuPercent = ((cpuDelta / systemDelta) * cpuCount * 100.0).toFixed(0);
+  // now log as appropriate.  verbose mode just always logs usage stats:
+  if (options.verbose) {
```

Comment:
I think we should have a log function that checks the threshold and if no threshold and options.verbose, then do the log.

Reply:
basically just move this check into the logThresholdExceeded function and rename to logContainer